spec/bolt/cli_spec.rb

```diff
@@ -2475,44 +2475,108 @@ def stub_config(file_content = {})
+
+          cli = Bolt::CLI.new(%w[project init myproject])
+          cli.execute(cli.parse)
+
           expect(File.file?(file)).to be
+          expect(File.read(file)).to eq("---\nname: myproject\n")
         end
       end
 
-      it 'warns when a bolt.yaml already exists' do
+      it 'errors if the directory name is invalid' do
         Dir.mktmpdir do |dir|
-          config = File.join(dir, 'bolt.yaml')
-          cli    = Bolt::CLI.new(%W[project init #{dir}])
+          allow(Dir).to receive(:pwd).and_return(dir)
+          cli = Bolt::CLI.new(%w[project init])
+          expect { cli.execute(cli.parse) }
+            .to raise_error(Bolt::ValidationError,
+                            /name '#{File.basename(dir)}' will be used as the project name, and is invalid/)
+        end
+      end
+
+      it 'errors if you pass in an invalid name' do
+        cli = Bolt::CLI.new(%w[project init 123])
+        expect { cli.execute(cli.parse) }
+          .to raise_error(Bolt::ValidationError, /The provided project name '123' is invalid;/)
+      end
+
+      it 'warns when a bolt-project.yaml already exists' do
+        Dir.mktmpdir do |tmpdir|
+          dir = File.join(tmpdir, 'valid')
+          Dir.mkdir(dir)
+          allow(Dir).to receive(:pwd).and_return(dir)
 
+          config = File.join(dir, 'bolt-project.yaml')
           FileUtils.touch(config)
+          cli = Bolt::CLI.new(%w[project init])
           cli.execute(cli.parse)
 
           expect(@log_output.readlines).to include(/Found existing project directory at #{dir}/)
         end
       end
 
+      it 'moves bolt.yaml to bolt-project.yaml if there are no inventory keys' do
+        Dir.mktmpdir do |tmpdir|
+          allow(Dir).to receive(:pwd).and_return(tmpdir)
+
+          old_config = File.join(tmpdir, 'bolt.yaml')
+          File.write(old_config, { 'color' => true }.to_yaml)
+
+          config = File.join(tmpdir, 'bolt-project.yaml')
+          cli    = Bolt::CLI.new(%w[project init valid])
+          cli.execute(cli.parse)
+
+          expect(File.file?(config)).to be
+          expect(File.file?(old_config)).not_to be
+          expect(File.read(config)).to eq("---\ncolor: true\n")
+        end
+      end
+
+      it 'errors if bolt.yaml exists and has inventory keys' do
+        Dir.mktmpdir do |tmpdir|
+          allow(Dir).to receive(:pwd).and_return(tmpdir)
+
+          old_config = File.join(tmpdir, 'bolt.yaml')
+          File.write(old_config, { 'ssh' => { 'host-key-check' => false } }.to_yaml)
+
+          cli = Bolt::CLI.new(%w[project init valid])
+          expect { cli.execute(cli.parse) }
+            .to raise_error(Bolt::Error, /Could not move #{old_config} to bolt-project.yaml/)
+        end
+      end
+
       context 'with modules' do
-        it 'creates a Puppetfile and installs modules with dependencies' do
+        it 'creates a Puppetfile and installs modules with dependencies', ssh: true do
```

Comment:
I'm still debugging why this is failing in CI, but it works manually on Windows so...I suspect it's a CI issue